src/parseMessageDefinition.js

```diff
@@ -190,20 +191,29 @@ export function parseMessageDefinition(messageDefinition: string) {
     .map((line) => line.trim())
     .filter((line) => line);
 
-  let definitionLines: string[] = [];
+  let definitionLines: { isJson: boolean, line: string }[] = [];
   const types: RosMsgDefinition[] = [];
+  let nextDefinitionIsJson: boolean = false;
   // group lines into individual definitions
   allLines.forEach((line) => {
-    // skip comment lines
-    if (line.indexOf("#") === 0) {
+    // ignore comment lines unless they start with #pragma rosbag_parse_json
+    const pragmaCommentIdx = line.indexOf("#pragma rosbag_parse_json");
+    if (line.indexOf("#") === 0 && pragmaCommentIdx !== 0) {
       return;
     }
+
     // definitions are split by equal signs
     if (line.indexOf("==") === 0) {
+      nextDefinitionIsJson = false;
       types.push(buildType(definitionLines));
       definitionLines = [];
     } else {
-      definitionLines.push(line);
+      definitionLines.push({ isJson: nextDefinitionIsJson, line });
+      nextDefinitionIsJson = false;
+    }
+
+    if (pragmaCommentIdx === 0) {
+      nextDefinitionIsJson = true;
```

Comment:
This code looks like it would push a definition for the line that has the pragma on it, that doesn't seem right?

Should we instead change a few lines above to:

```js
if (line.startsWith("#")) {
  if (line.startsWith("#pragma rosbag_parse_json")) {
    nextDefinitionIsJson = true;
  }
}
```

(While I was at it, I changed `indexOf(…) === 0` to `startsWith(…)` since the former is a bit old-skool / less readable.)

Reply:
Ah yeah - makes sense. I added `startsWith` elsewhere in this function as well.